packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -0,0 +1,84 @@
+import { version as nodeVersion } from 'node:process'
+
+import { getDeployStore } from '@netlify/blobs'
+import pMap from 'p-map'
+import semver from 'semver'
+
+import { addErrorInfo } from '../../error/info.js'
+import { log, logError } from '../../log/logger.js'
+import { getBlobsDir } from '../../utils/blobs.js'
+
+import { getKeysToUpload, getFileWithMetadata, anyBlobsToUpload } from './utils.js'
+
+const coreStep = async function ({
+  debug,
+  logs,
+  deployId,
+  buildDir,
+  quiet,
+  constants: { PUBLISH_DIR, SITE_ID, NETLIFY_API_TOKEN, API_URL },
+}) {
+  const storeOpts: { siteID: string; deployID: string; token: string; apiURL: string; fetch?: any } = {
+    siteID: SITE_ID,
+    deployID: deployId,
+    token: NETLIFY_API_TOKEN,
+    apiURL: API_URL,
+  }
+  if (semver.lt(nodeVersion, '18.0.0')) {
+    const nodeFetch = await import('node-fetch')
+    storeOpts.fetch = nodeFetch.default
+  }
+
+  const blobStore = getDeployStore(storeOpts)
+  const blobsDir = getBlobsDir({ buildDir, publishDir: PUBLISH_DIR })
+  const keys = await getKeysToUpload(blobsDir)
+
+  // We checked earlier, but let's be extra safe
+  if (keys.length === 0) {
+    if (!quiet) {
```

Comment:
> is this because we'll call this "manually" from the CLI and we want to make sure we don't log anything in that case?

Correct, in case we call it with `--json` or similar flags (in fact, I found out about this when working on the CLI tests that used `--json`)

> If so and this is a tiny little improvement, making this a wrapper that moves away the constant quiet check could be a nice + for readibility, what do you think?

I thought about that and asn't sure though if to make it in this package or in the general logger (in fact, I was surprised `log` didn't take into account the `quiet` check). I'll start by making it in this package only, but happy to change to a more global one.

Reply:
Will leave this for a follow-up as we have something similar in other places already and I prefer consistency, and don't wanna increase the scope of this.